react/components/SearchResult.js

```diff
@@ -39,15 +39,15 @@ class SearchResult extends Component {
     hiddenFacets: this.props.hiddenFacets,
   }
 
-  handleLayoutChange = e => {
+  handleMobileLayoutChange = e => {
     e.preventDefault()
 
-    const defaultModes = [this.props.hiddenFacets.layoutMode1, this.props.hiddenFacets.layoutMode2]
-    const modeIndex = (defaultModes.indexOf(this.state.galleryLayoutMode) + 1) % 2
+    const defaultModes = [this.props.mobileLayout.mode1, this.props.mobileLayout.mode2]
```

Comment:
I'm not sure if defaultModes makes sense here, maybe only 'modes' would be more descriptive and avoid misunderstandings

Reply:
Fixed.